client/src/components/Review Widget/ProductBreakdown.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import './cssFiles/productBreakdown.css'
+import ProductBreakDownBar from './ProductBreakDownBar.jsx';
+
+
+const ProductBreakdown = (props) => {
+
+  if (props === undefined) {
+    return null;
+  }
+
+  const getRatingName = (charName, valueType) => {
+    if (charName === 'Size') {
+      return (valueType === 'low' ? 'A size too small' : 'A size too wide');
+    } else if (charName === 'Width') {
+      return (valueType === 'low' ? 'A size too small' : 'A size too wide');
+    } else if (charName === 'Comfort') {
+      return (valueType === 'low' ? 'Uncomfortable' : 'Perfect');
+    } else if (charName === 'Quality') {
+      return (valueType === 'low' ? 'Poor' : 'Perfect');
+    } else if (charName === 'Length') {
+      return (valueType === 'low' ? 'Runs short' : 'Runs long');
+    } else {
+      return (valueType === 'low' ? 'Runs tight' : 'Runs long');
+    }
+  }
+
+  let breakDownComponents = [];
+
+  for (var key in props.characteristicsData) {
+    breakDownComponents.push(
```

Comment:
As a suggestion, since you are already doing a lot of html in this loop. You might make this a component of it's own and call it on loop